Sources/Components/Broadcast/BroadcastContainerView/BroadcastContainerView.swift

```diff
@@ -0,0 +1,258 @@
+                UIView.animate(withDuration: 0.3, animations: { [weak self] in
+                    self?.contentView.arrangedSubviews.forEach { $0.isHidden = false }
+                })
+            })
+        } else {
+            UIView.animate(withDuration: 0.3, animations: { [weak self] in
+                self?.contentView.arrangedSubviews.forEach { $0.isHidden = false }
+            })
+        }
+    }
+
+    public override var intrinsicContentSize: CGSize {
+        guard contentView.arrangedSubviews.count != 0 else {
+            return CGSize(width: frame.size.width, height: 0)
+        }
+
+        let broadcastViewsHorizontalSpacings = LayoutConstants.ContentView.insets.leading + abs(LayoutConstants.ContentView.insets.trailing)
+        let constrainedWidth = frame.size.width - broadcastViewsHorizontalSpacings
+
+        let broadcastViewsTotalHeight = contentView.arrangedSubviews.reduce(CGFloat(0), { acc, view in
+            guard let broadcastView = view as? BroadcastView else {
+                return acc
+            }
+
+            let broadcastViewHeight = broadcastView.calculatedSize(withConstrainedWidth: constrainedWidth).height
+
+            return acc + broadcastViewHeight
+        })
+
+        guard broadcastViewsTotalHeight != 0 else {
+            return CGSize(width: frame.size.width, height: 0)
+        }
+
+        let broadcastViewTotalSpacing = ((CGFloat(contentView.arrangedSubviews.count) * LayoutConstants.BroadcastView.spacing) - .mediumLargeSpacing)
+        let verticalSpacing = LayoutConstants.ContentView.insets.top + broadcastViewTotalSpacing
+        let containerHeight = broadcastViewsTotalHeight + verticalSpacing
+
+        return CGSize(width: frame.size.width, height: containerHeight)
+    }
+}
+
+// MARK: - Private
+
+private extension BroadcastContainerView {
+    func setup() {
+        contentView.frame = CGRect(x: 0, y: 0, width: frame.width, height: 0)
+
+        addSubview(contentView)
+
+        let insets = LayoutConstants.ContentView.insets
+
+        NSLayoutConstraint.activate([
+            contentView.topAnchor.constraint(equalTo: topAnchor, constant: insets.top),
+            contentView.leadingAnchor.constraint(equalTo: leadingAnchor, constant: insets.leading),
+            contentView.trailingAnchor.constraint(equalTo: trailingAnchor, constant: insets.trailing),
+        ])
+
+        contentView.setContentCompressionResistancePriority(.defaultLow, for: .vertical)
+        contentView.setContentHuggingPriority(.defaultHigh, for: .vertical)
+    }
+
+    func layoutBroadcastViews(from broadcasts: [Broadcast]) {
+        let broadcastViews = broadcasts.map { broadcastView(from: $0) }
+
+        broadcastViews.forEach { add($0, to: contentView) }
+    }
+
+    func broadcastView(from broadcast: Broadcast) -> BroadcastView {
+        let broadcastView = BroadcastView(frame: .zero)
+
+        // Set the message on the broadcast view and prepare it in its expanded state
```

Comment:
I don't think this comment is needed. Comments should say "why" not "what". :)

Reply:
removed